### PR TITLE
Time matching algorithm

### PR DESCRIPTION
**Pull request by stephend017**
Implemented and rigorously tested the base version of the time matching algorithm. This can handle unlimited number of participants and allows for a minimum meeting length. Please look through test cases to make sure all cases are tested

### CODE REVIEW

File: biit_server/time_matcher.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+from typing import Tuple, List
+
+
+class AvailabilityManager:
+    """
+    Class for managing the total availability of a
+    given number of participants
+    """
+
+    def __init__(self, people: List[Tuple[int, int]], meeting_length: int = 0):
```
Comment (lzhahn) [inline biit_server/time_matcher.py]: Add a docstring for what each of the variables represents

**Review by lzhahn — APPROVED** [review]
Looks good, needs a docstring on the AvalibilityManager then it's good with me

**Review by lzhahn — COMMENTED** [review]
LGTM! The only thing is a test for times on different day i.e. users are available on multiple days, other than that send it!

**Comment by stephend017** [comment]
> LGTM! The only thing is a test for times on different day i.e. users are available on multiple days

I have a solution to that. we just add 24 for each day. so monday is 0 - 23, tuesday is 24 - 47 etc... the algorithm will still work and nothing needs to be changed, only the inputs.

**Review by fatcat2 — APPROVED** [review]
looks ok

**Pull request merged**